java/AsyncIO/src/main/java/com/amazonaws/services/msf/RetriesFlinkJob.java

```diff
@@ -0,0 +1,110 @@
+                                                        .getResource(LOCAL_APPLICATION_PROPERTIES_RESOURCE).getPath());
+                } else {
+                        LOGGER.debug("Loading application properties from Amazon Managed Service for Apache Flink");
+                        return KinesisAnalyticsRuntime.getApplicationProperties();
+                }
+        }
+
+        public static void main(String[] args) throws Exception {
+
+                final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+                // Load application parameters
+                final Map<String, Properties> applicationParameters = loadApplicationProperties(env);
+
+                DataStreamSource<IncomingEvent> source = env.fromSource(
+                                getIncomingEventGeneratorSource(), WatermarkStrategy.noWatermarks(), "data-generator")
+                                .setParallelism(1);
+
+                // Defining fixed delay retry strategy
+                AsyncRetryStrategy retryStrategy = new AsyncRetryStrategies.FixedDelayRetryStrategyBuilder<ProcessedEvent>(
+                                3, 1000) // maxAttempts=3, initialDelay=1000 (in ms)
+                                .ifResult(RetryPredicates.EMPTY_RESULT_PREDICATE)
+                                .ifException(RetryPredicates.HAS_EXCEPTION_PREDICATE)
+                                .build();
+
+                // Async I/O transformation with retry
+                DataStream<ProcessedEvent> processedStream = AsyncDataStream.unorderedWaitWithRetry(
+                                source,
+                                new ProcessingFunction(applicationParameters.get("EndpointService")),
+                                2000, // Timeout
+                                TimeUnit.MILLISECONDS,
+                                100, // Capacity
+                                retryStrategy);
+
+                processedStream
+                                .map(value -> String.format("%s,%s", value.message, value.processed))
+                                .sinkTo(createSink(applicationParameters.get("OutputStream")));
+
+                LOGGER.debug("Starting flink job: {}", "Async I/O Retries");
+
+                env.execute("Async I/O Job");
+        }
+
+        private static DataGeneratorSource<IncomingEvent> getIncomingEventGeneratorSource() {
+                long recordPerSecond = 1;
+                return new DataGeneratorSource<>(
+                                new IncomingEventDataGeneratorFunction(),
+                                Long.MAX_VALUE,
+                                RateLimiterStrategy.perSecond(recordPerSecond),
+                                TypeInformation.of(IncomingEvent.class));
+        }
+
+        private static KinesisStreamsSink<String> createSink(Properties outputProperties) {
+                String outputStreamName = outputProperties.getProperty("stream.name");
```

Comment:
Parameter is called stream name but is stream ARN

Reply:
Probably better passing the name, for consistency with other examples. Also simpler to put a generic name in `flink-application-properties-dev.json`

Reply:
setStreamName was recently deprecated. We changed the naming to be consistent with stream ARN